### ISSUE TITLE
Revise map_reduce() interface

### ISSUE THREAD

**Issue by jdgarciauc3m** [arcosuc3m/grppi#122]
`map_reduce()` should take initial value as last argument.

~~~c++
auto res = map_reduce(ex, begin(v), end(v), 
  [](auto x) { return x*x; },
  [](auto x, auto y) { return x+y; },
  0.0
);

auto res = map_reduce(ex, begin(v), end(v), 
  [](auto x) { return x*x; }
  [](auto x, auto y) { return x+y; }
); // Assumes initial value to 0.0

~~~

**Comment by jdgarciauc3m** [comment]
Besides.

Does it make sense to have a `map_reduce` version with an output iterator for the result?

**Comment by jdgarciauc3m** [comment]
Solved by PR #194
